project-chip/connectedhomeip
# Bridge app test updates

#### Problem
The bridge-app example was updated to support the bridge test plan and to conform to specification changes.

#### Change overview
Commands were added to the bridge-app to add, remove, rename, and change the state for dynamic endpoints with a Matter aggregator(bridge).
The aggregator(bridge) was changed to use a non-zero endpoint ID to match the specification.
The bridge-app was changed so that it does not use fixed labels. (A function replaced by Action clusters).

#### Testing
The changes to the bridge-app were tested by running the chip-bridge-app from a Linux terminal and using chip-tool to examine the parts-list, node labels, and on-off state.

1. Start bridge app and wait for initialization to complete
2. Run chip-tool pairing
chip-tool pairing ethernet 12344321 20202021 3840 localhost 5540
3. Run chip-tool to get parts-list and verify all endpoint IDs are listed
chip-tool descriptor read parts-list 12344321 0
4. Run chip-tool to get parts-list for the aggregator(bridge) and verify bridged nodes are listed
chip-tool descriptor read parts-list 12344321 1
5. Press 2  in bridge-app to add "light 2"
6. Run chip-tool to observe a new endpoint in parts-list
chip-tool descriptor read parts-list 12344321 1
7. Press 4 in the bridge-app to remove "Light 1"
8. Run chip-tool to observe "Light 1" was removed from parts-list
chip-tool descriptor read parts-list 12344321 1
9. Press 5 in the bridge-app to add "Light 1"
10. Run chip-tool to observe "Light 1" was added again (it will have a different endpoint number)
chip-tool descriptor read parts-list 12344321 1
11. Run chip-tool to observe the node label for "Light 1"
./out/host/chip-tool bridgeddevicebasic read node-label 12344321 11
12. Press b in the bridge-app to rename Light 1
13. Run chip-tool to observe the node label for "Light 1" changed to "Light 1b"
14. Run chip-tool to observe on-off state for "Light 1" to see initial on-off state
chip-tool onoff read  on-off 12344321 11
15. Press c in the bridge-app to change the on-off state of the lights
16. Run chip-tool to observe the on-off state for "Light 1" has changed
chip-tool onoff read  on-off 12344321 11
